### DISCUSSION TITLE
Broken and stale unit tests

### DISCUSSION THREAD

**Discussion by dobanisola-scottlogic** [primefaces/community#1812] [PrimeNG] [answered]
I currently ran the unit tests on the project while I was fixing a bug and noticed that there's quite a number of broken unit tests. I'm considering contributing to resolving them if possible. I'll like to know if there's any reason the tests were left broken and if fixing the test is a good use of my time. Also, is there any consideration to migrate from karma to jest?

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

During the implementation of accessibility in v16, we had to rewrite many components, which left some tests invalid. Our goal for v18 is to strengthen the tests. We are grateful for your willingness to contribute. Contributions like these greatly help us advance PrimeNG. If you wish to submit a PR and make corrections, please feel free to do so. However, please submit a PR for one component first, and let's review it, don't forget to open an issue first and link the issue with the PR. You can also add some cases that we missed to extend the test coverage.

Thanks in advance!

**Reply by dobanisola-scottlogic** [reply]
Hi,

Please see the link to the PR as requested.
https://github.com/primefaces/primeng/pull/15493

Thanks

**Reply by alexciesielski** [reply]
I noticed the same during work on my PR and opened this issue as well: https://github.com/primefaces/primeng/issues/15466

**Comment by cetincakiroglu** [comment]
Hi all,

We appreciate the effort and support received from you for broken tests to make PrimeNG better, you guys rock!

**Reply by dobanisola-scottlogic** [reply]
Thank you so much for your kind words! We greatly appreciate your support and recognition. I'm happy to report that our team - [Rob](https://github.com/robGardiner01), [Lisa](https://github.com/RogueTea), [Nathanael](https://github.com/nbrown-ScottLogic) and I - have successfully fixed several issues and all the broken unit tests except for one, which intermittently passes and fails. We wanted to bring this to your attention as it may require further investigation.

While we will be reducing our contributions to the project moving forward, it has been a truly rewarding experience working on this project. We have learned a lot and are proud to have contributed to its improvement.

Thank you once again for the opportunity.

Best regards,
